MCprep_addon/spawner/spawn_util.py

```diff
@@ -41,11 +45,298 @@
+			return [f"zombie/{self.zombie_variation.lower()}.png"]
+		elif mob_type == "Skeleton":
+			return [f"zombie/{self.skeleton_variation.lower()}.png"]
+		elif mob_type == "Allay" or mob_type == "Vex":
+			return [] if self.is_zombiefied or mob_type == "Vex" else []
+			# TODO: Add the rest of the mobs
+			
+	profession_variation: bpy.props.EnumProperty(name="Profession", items=profession_items)
+	level_variation: bpy.props.EnumProperty(name="Level", items=level_items)
+	biome_variation: bpy.props.EnumProperty(name="Biome", items=biome_items)
+	zombie_variation: bpy.props.EnumProperty(name="Zombie Variation", items=zombie_items)
+	skeleton_variation: bpy.props.EnumProperty(name="Skeleton Variation", items=skeleton_items)
+	is_zombiefied: bpy.props.BoolProperty(name="Is Zombiefied")  # Use this for Allay-Vex
+	fox_variation: bpy.props.EnumProperty(items=fox_items)
+	dog_variation: bpy.props.EnumProperty(items=dog_items)
+	cat_variation: bpy.props.EnumProperty(items=cat_items)
+	
+	def draw_variation_ui(self, context: Context, layout: UILayout):
+		""" Sharable method for drawing"""
+		obj = context.object
+		mob_type = getmob_type(obj)
+		if mob_type in ["Villager", "Trader"]:
+			layout.prop(self, "profession_variation")
+			layout.prop(self, "level_variation")
+			layout.prop(self, "biome_variation")
+			
+		elif mob_type == "Zombie":
+			layout.prop(self, "zombie_variation")
+		elif mob_type == "Skeleton":
+			layout.prop(self, "skeleton_variation")
+		elif mob_type == "Axolotl":
+			layout.prop(self, "axolotl_variation")
+		elif mob_type == "Rabbit":
+			layout.prop(self, "rabbit_variation")
+		elif mob_type == "Frog":
+			layout.prop(self, "frog_variation")
+		elif mob_type == "Parrot":
+			layout.prop(self, "parrot_variation")
+		elif mob_type == "Llama":
+			layout.prop(self, "llama_variation")
+		elif mob_type == "Horse":
+			layout.prop(self, "horse_variation")
+		elif mob_type == "Cat" or mob_type == "Ocelot":
+			layout.prop(self, "cat_variation")
+		elif mob_type == "Dog" or mob_type == "Wolf":
+			layout.prop(self, "dog_variation")
+			
+		counter_variant = ["Villager", "Piglin", "Hoglin", "Allay", "Vex"]
+		if mob_type in counter_variant:
+			text = "Is Vex" if mob_type == "Allay" else "Is Zombified"
+			layout.prop(self, "is_zombiefied", text=text)
+		
+		has_variant = counter_variant + ["Zombie", "Skeleton", "Llama", "Axolotl", "Rabbit", "Llama", "Parrot", "Frog", "Horse", "Cat", "Ocelot"]
+		if mob_type == "Custom" or mob_type not in has_variant:
+			layout.label(text="This mob doesn't has any variant to swap skin yet")
+		elif mob_type == "Player":
+			layout.label(text="Please use Player for this")
+
 # -----------------------------------------------------------------------------
 # Reusable functions for spawners
 # -----------------------------------------------------------------------------
 
 
+def getmob_type(rig: bpy.types.Object):
+	""" Get mob type from rig
+	args
+		obj: Armature Object
+	"""
```

Comment:
I'd also add some documentation on what this function returns (don't forget the return value type as well), and what values it returns in what conditions
```suggestion
def getmob_type(rig: bpy.types.Object) -> bool:
	""" Get mob type from rig
	args
		rig: Armature Object
		
	Returns:
		True if (insert condition), False otherwise
	"""
```